htdocs/fourn/class/fournisseur.product.class.php

```diff
@@ -989,29 +1005,67 @@ public function getSocNomUrl($withpicto = 0, $option = 'supplier', $maxlen = 0,
 	public function display_price_product_fournisseur($showunitprice = 1, $showsuptitle = 1, $maxlen = 0, $notooltip = 0, $productFournList = array())
 	{
 		// phpcs:enable
-		global $conf, $langs;
+		global $conf, $langs, $hookmanager;
 
 		$out = '';
 		$langs->load("suppliers");
-		if (count($productFournList) > 0) {
-			$out .= '<table class="nobordernopadding" width="100%">';
-			$out .= '<tr><td class="liste_titre right">'.($showunitprice ? $langs->trans("Price").' '.$langs->trans("HT") : '').'</td>';
-			$out .= '<td class="liste_titre right">'.($showunitprice ? $langs->trans("QtyMin") : '').'</td>';
-			$out .= '<td class="liste_titre">'.$langs->trans("Supplier").'</td>';
-			$out .= '<td class="liste_titre">'.$langs->trans("SupplierRef").'</td></tr>';
-			foreach ($productFournList as $productFourn) {
-				$out .= '<tr><td class="right">'.($showunitprice ? price($productFourn->fourn_unitprice * (1 - $productFourn->fourn_remise_percent / 100) - $productFourn->fourn_remise) : '').'</td>';
-				$out .= '<td class="right">'.($showunitprice ? $productFourn->fourn_qty : '').'</td>';
-				$out .= '<td>'.$productFourn->getSocNomUrl(1, 'supplier', $maxlen, $notooltip).'</td>';
-				$out .= '<td>'.$productFourn->fourn_ref.'<td></tr>';
-			}
-			$out .= '</table>';
+
+		// VSPR  Because calc is hardcoded we need a hook to override price display. Based on the Extra Discounts Addon adding 4 more discounts to supplier
+		global $action;
+		$hookmanager->initHooks(array($this->element . 'dispprice'));
+		$parameters = array('showunitprice'=>$showunitprice,
+			'showsuptitle'=>$showsuptitle,
+			'maxlen'=>$maxlen,
+			'thisid'=>$this->id,
+			'thisproductfournpriceid'=>$this->product_fourn_price_id,
+			'thisfournunitprice'=>$this->fourn_unitprice,
+			'thisfournremisepercent'=>$this->fourn_remise_percent,
+			'thisfournremise'=>$this->fourn_remise,
+			'thissupplier'=>$this->getSocNomUrl(1, 'supplier', $maxlen, $notooltip), 
+			'thisrefsupplier'=>$this->ref_supplier,
+			'notooltip'=>$notooltip
+		);
+		$reshook = $hookmanager->executeHooks('displaypriceproductfournisseur', $parameters, $productFournList, $action); // Note that $action and $object may have been modified by some hooks
+		if ($reshook > 0) {
+			$result = $hookmanager->resPrint;
 		} else {
-			$out = ($showunitprice ? price($this->fourn_unitprice * (1 - $this->fourn_remise_percent / 100) + $this->fourn_remise, 0, $langs, 1, -1, -1, $conf->currency).' '.$langs->trans("HT").' &nbsp; <span class="opacitymedium">(</span>' : '');
-			$out .= ($showsuptitle ? '<span class="opacitymedium">'.$langs->trans("Supplier").'</span>: ' : '').$this->getSocNomUrl(1, 'supplier', $maxlen, $notooltip).' / <span class="opacitymedium">'.$langs->trans("SupplierRef").'</span>: '.$this->ref_supplier;
-			$out .= ($showunitprice ? '<span class="opacitymedium">)</span>' : '');
+			$result .= $hookmanager->resPrint;
```

Comment:
I think you can remove the else case because you only want to replace something
![image](https://github.com/Dolibarr/dolibarr/assets/3624836/62216cee-2fa3-4f2c-889a-87e59918f483)

Reply:
I remove that one and changed other things, but this is really strange. my vim is set to 4 spaces for a tab and also set list shows 3 tabs on 1148:

^I^I^I$this->db->free($resql);$

But this precheck thinks it founds 2 on 1148.

Check failure on line 1148 in htdocs/fourn/class/fournisseur.product.class.php
GitHub Actions
/ pre-commit / pre-commit

Line indented incorrectly; expected at least 3 tabs, found 2

I give up for today and with loosing so much hours for tabs/whitespaces and error messages showing strange errors in areas i have not made changes I think i give up with this PR and keep my changes local, patching the files if a update happens. Maybe i need to blame myself to not get this done, but I need to move my modules forward and can not loose so much hours for tabs. :-)

Will sleep a night over this.